pyiceberg/io/pyarrow.py

```diff
@@ -1068,20 +1068,13 @@ def primitive(self, primitive: pa.DataType) -> PrimitiveType:
             return StringType()
         elif pa.types.is_date32(primitive):
             return DateType()
-        elif isinstance(primitive, pa.Time64Type) and primitive.unit == "us":
+        elif isinstance(primitive, pa.Time64Type) and primitive.unit in ["us", "ns"]:
```

Comment:
dont need to check for precision here, since its `us` and `ns` are supported.

Although we need to add code to downcast `ns` to `us`, similar to below

Reply:
> dont need to check for precision here, since its us and ns are supported.

[Time64Type](https://arrow.apache.org/docs/python/generated/pyarrow.Time64Type.html) only supports `us` and `ns`, so the extra check for `primitive.unit in ["us", "ns"]` is redundant. Although, it'll be helpful if another precision was added in the future.

> Although we need to add code to downcast ns to us, similar to below

Because Iceberg doesn't support `ns`, it has to downcast `ns` to `us`, or errors. 
This behavior is controlled by https://github.com/apache/iceberg-python/pull/1206/files#diff-8d5e63f2a87ead8cebe2fd8ac5dcf2198d229f01e16bb9e06e21f7277c328abdL1078-L1084

Which we also need to include here

Reply:
Like this?
```
        elif pa.types.is_timestamp(primitive):
            primitive = cast(pa.TimestampType, primitive)
            if primitive.unit in ("s", "ms", "us"):
                # Supported types, will be upcast automatically to 'us'
                pass
            elif primitive.unit == "ns":
                if self._downcast_ns_timestamp_to_us:
                    primitive = cast(pa.TimestampType, "us")
                else:
                    raise TypeError(
                        "Iceberg does not yet support 'ns' timestamp precision. Use 'downcast-ns-timestamp-to-us-on-write' configuration property to automatically downcast 'ns' to 'us' on write."
                    )
```

Reply:
yep, do the same for `Time64Type`

Reply:
```
            elif primitive.unit == "ns":
                if self._downcast_ns_timestamp_to_us:
                    primitive = cast(pa.TimestampType, "us")
                    return TimeType(primitive)
```

Reply:
You mean return new Timetype using new primitive.unit right?

Reply:
```
            elif primitive.unit == "ns":
                if self._downcast_ns_timestamp_to_us:
                    primitive = cast(pa.TimestampType, "us")
                    primitive = cast(pa.Time64Type, "us")
```

Reply:
```
            elif primitive.unit == "ns":
                if self._downcast_ns_timestamp_to_us:
                    primitive = cast(pa.TimestampType, "us")
                    cast(pa.Time64Type, "us")
```

Reply:
You mean this?
```
            elif primitive.unit == "ns":
                if self._downcast_ns_timestamp_to_us:
                    primitive = cast(pa.Time64Type, primitive)
```

Reply:
the `_downcast_ns_timestamp_to_us` part for sure. Im not sure if we need to do `cast` though

Reply:
I am confused. 
Did you mean that I need a new class variable _downcast_ns_time64type_to_us and a new condition to check?

```
if self._downcast_ns_time64type_to_us:
    # what should be done here?
```

Or should I add some action under the original condition:
```
if self._downcast_ns_timestamp_to_us:
    # Add something here. 
```

Reply:
So this function is converting pyarrow types to Iceberg types. 

In the example of timestamp type (`pa.types.is_timestamp(primitive)`), we check if the `unit` (or precision) is `ns`. If it is and the `downcast-ns-timestamp-to-us-on-write` is set, we want to downcast the `unit` from `ns` to `us`.
If its not, we don't do anything special and just return the Iceberg type.

We want to do the same thing with the `Time64Type` type.

Reply:
essentially check if the `unit` is `ns`. if it is, check `downcast-ns-timestamp-to-us-on-write`. if the option is set, we can downcast from `ns` to `us`. 
If its not, raise an error

Reply:
Please check the new commit. Thanks.